keyboards/helix/helix.c

```diff
@@ -0,0 +1,153 @@
+/* Copyright 2020 yushakobo
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "helix.h"
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case EISU:
+            if (record->event.pressed) {
+                if (is_mac_mode()) {
+                    register_code(KC_LNG2);
+                } else {
+                    tap_code16(LALT(KC_GRAVE));
+                }
+            } else {
+                unregister_code(KC_LNG2);
+            }
+            return false;
+            break;
+        case KANA:
+            if (record->event.pressed) {
+                if (is_mac_mode()) {
+                    register_code(KC_LNG1);
+                } else {
+                    tap_code16(LALT(KC_GRAVE));
+                }
+            } else {
+                unregister_code(KC_LNG1);
+            }
+            return false;
+            break;
+        case RGBRST:
+#ifdef RGBLIGHT_ENABLE
+            if (record->event.pressed) {
+                eeconfig_update_rgblight_default();
+                rgblight_enable();
+            }
+#endif
+            break;
+    }
+    return process_record_user(keycode, record);
+}
+
+bool is_mac_mode(void) {
+    return keymap_config.swap_lalt_lgui == false;
+}
+
+void set_mac_mode(bool macmode) {
+    keymap_config.swap_lalt_lgui = keymap_config.swap_ralt_rgui = !macmode;
+    eeconfig_update_keymap(keymap_config.raw);
+}
+
+#ifdef DIP_SWITCH_ENABLE
+bool dip_switch_update_kb(uint8_t index, bool active) {
+    if (!dip_switch_update_user(index, active)) {
+        return false;
+    }
+    switch (index) {
+        case 0:
+            active ? set_mac_mode(false) : set_mac_mode(true);
+            break;
+    }
+    return true;
+}
+#endif
+
+#ifdef OLED_ENABLE
+void render_status(void) {
+    static const char os_logo[][2][3] PROGMEM = {{{0x95, 0x96, 0}, {0xb5, 0xb6, 0}}, {{0x97, 0x98, 0}, {0xb7, 0xb8, 0}}};
```

Comment:
You could probably just do: 
```suggestion
    static const char os_logo[][5] PROGMEM = {{0x95, 0x96, '\n', 0x97, 0x98, 0}, {0xb5, 0xb6, '\n', 0xb7, 0xb8, 0}};
```

Don't think it would save any space, but would be a bit cleaner.

Would also allow you to just do:
```c
oled_write_P(os_logo[is_mac_mode()], false);
```